Паттерн MVVM

TemZZZ opened the pull request.

Commit:
feature(project MVVM, LoadFromFileViewModel.ActionCommand): добавлен вывод сообщений об ошибке или успешности загрузки радиокомпонентов

Commit:
style(project MVVM, LoadFromFileViewModel.ActionCommand): изменен XML комментарий

Commit:
feture(project MVVM, SaveToFileViewModel.GetWritingRadiocomponents): добавлен выброс исключения при передаче в метод null

Commit:
feture(project MVVM, MainViewModel.ToPrintableRadiocomponents): добавлен выброс исключения при передаче в метод null

Commit:
feature(project MVVM, MainViewModel.OpenSaveToFileWindowCommand): добавлена проверка на null SelectedObjects

Commit:
fix(project MVVM, IDialogService): исправлены XML комментарии методов OpenFileDialog и SaveFileDialog

- была не верна информация о возвращаемых значениях

Commit:
refactor+style(project MVVM, SaveToFileViewModel): класс переписан как наследник ActionViewModelBase

- также добавлены XML комментарии к членам класса

Commit:
refactor(project MVVM, LoadFromFileViewModel.ActionCommand): метод ReadXmlAndDeserialize теперь вызывается в дополнительным параметром openFileDialog.ShowMessage

- удалено отображение информационного диалогового окна с сообщением loadingErrorMessage

Commit:
refactor(project MVVM, SaveToFileViewModel.ActionCommand): метод SerializeAndWriteXml теперь вызывается в дополнительным параметром openFileDialog.ShowMessage

Commit:
feature(project MVVM, MainWindow.xaml): немного изменены параметры окна

- изменен заголовок на Коллекция радиокомпонентов
- удалено свойство SizeToContent окна
- заданы Height и Width окна, равными соответствующим минимальным значениям
- свойству ColumnWidth элемента CustomDataGrid задано значение *

Commit:
style(project Model, RadiocomponentToPrintableRadiocomponentAdapter.GetRadiocomponent): изменен XML комментарий

Commit:
feature(project MVVM): добавлена разметка окна поиска радиокомпонентов SearchRadiocomponentWindow

Commit:
feature(project MVVM, SearchRadiocomponentWindow.xaml): реализовано закрытие окна кнопкой Отмена

Commit:
feature(project MVVM): добавлен класс SearchRadiocomponentViewModel без реализации

Commit:
feature(project MVVM, App.xaml.cs, RegisterWindowsTypes): зарегистрирован класс SearchRadiocomponentViewModel на тип окна SearchRadiocomponentWindow

Commit:
feature(project MVVM, MainViewModel): реализовано открытие окна поиска радиокомпонентов кликом командой OpenSearchWindowCommand

Commit:
feature(project MVVM, SearchRadiocomponentViewModel): добавлен конструктор с параметрами

- передаются коллекция радиокомпонентов, по которым производится поиск, и допустимые типы для поиска

Commit:
feature(project MVVM, SearchRadiocomponentWindow.xaml): добавлен вылидатор NotNegativeDoubleValidationRule

- правило валидации назначено текстовым поля для ввода ограничений фильтров "больше, чем", "меньше, чем", "равно"
- фитры привязаны  к свойствам MoreThanThresholdAsString, LessThanThresholdAsString, EqualsThresholdAsString

Commit:
feature(project MVVM, SearchRadiocomponentWindow.xaml): чекбоксы теперь активируют/деактивируют поля ввода параметров фильтров

Commit:
feature(project MVVM, MainViewModel.OpenSearchWindowCommand): в конструктор SearchRadiocomponentViewModel передаются _availableRadiocomponentTypes, Radiocomponents

Commit:
feature(project MVVM, SearchRadiocomponentViewModel): в конструкторе создается коллекция проиндексированных радиокомпонентов

- в классе реализован метод GetIndexedRadiocomponents
- в класс добавлено приватное поле _indexedRadiocomponents
- тип параметра radiocomponents изменен с IEnumerable на IList
- полю _indexedRadiocomponents присваивается значение GetIndexedRadiocomponents(radiocomponents)
- удалено приватное поле _radiocomponents

Commit:
feature(project MVVM, SearchRadiocomponentViewModel): в конструкторе создается словарь пар имя типа-тип радиокомпонента

- в классе реализован метод GetTypeNameToRadiocomponentTypeMap
- в класс добавлено приватное поле __typeNameToRadiocomponentTypeMap
- полю _typeNameToRadiocomponentTypeMap присваивается значение GetTypeNameToRadiocomponentTypeMap(availableRadiocomponentTypes)
- удалено приватное поле _availableRadiocomponentTypes

Commit:
feature(project MVVM, SearchRadiocomponentViewModel): добавлен и реализован метод фильтрации радиокомпонентов по типу GetFilteredByTypeIndexedRadiocomponents

Commit:
feature(project MVVM, SearchRadiocomponentViewModel): добавлен и реализован метод фильтрации радиокомпонентов по значению GetFilteredByValueIndexedRadiocomponentsIndices

Commit:
feature(project MVVM, SearchRadiocomponentViewModelюGetFilteredByTypeIndexedRadiocomponents): добавлена проверка параметра indexedRadiocomponents на null

Commit:
feature(project MVVM, SearchRadiocomponentViewModelюGetFilteredByTypeIndexedRadiocomponents): добавлен параметр включения/отключения компаратора isComparatorTurnedOn

- если false, то вернутся индексы всех элементов

Commit:
feature(project MVVM, SearchRadiocomponentViewModel): добавлены компараторы _lessThan, _moreThan, _equals

Commit:
feature(project MVVM, SearchRadiocomponentViewModel): добавлены приватные поля для отслеживания состояний фильтров

Commit:
feature(project MVVM, SearchRadiocomponentViewModel): добавлены публичные свойства IsLessThanFilterTurnedOn, IsMoreThanFilterTurnedOn, IsEqualsFilterTurnedOn

Commit:
refactor(project MVVM, SearchRadiocomponentViewModel): переименованы поля

- _lessThanFilterThresholdValid в _isLessThanFilterThresholdValid
- _moreThanFilterThresholdValid в _isMoreThanFilterThresholdValid
- _equalsFilterThresholdValid в _isEqualsFilterThresholdValid

Commit:
feature(project MVVM, SearchRadiocomponentViewModel): добавлены публичные свойства LessThanFilterThresholdAsString, MoreThanFilterThresholdAsString, EqualsFilterThresholdAsString

Commit:
feature(project MVVM, SearchRadiocomponentWindow.xaml): привязаны свойства IsMoreThanFilterTurnedOn, IsLessThanFilterTurnedOn, IsEqualsFilterTurnedOn вьюмодели SearchRadiocomponentViewModel к чекбоксам фильтров и свойства LessThanFilterThresholdAsString, MoreThanFilterThresholdAsString, EqualsFilterThresholdAsString к текстовым полям параметров фильтров

Commit:
fix(project MVVM, SearchRadiocomponentViewModel): исправлены модификаторы доступа свойств IsLessThanFilterTurnedOn, IsMoreThanFilterTurnedOn, IsEqualsFilterTurnedOn с private на public

Commit:
feature(project MVVM, SearchRadiocomponentViewModel): добавлено публичное свойство AvailableRadiocomponentTypesNames

Commit:
feature(project MVVM, SearchRadiocomponentWindow.xaml): свойство AvailableRadiocomponentTypesNames привязано к комбобоксу

Commit:
feature(project MVVM, SearchRadiocomponentViewModel): добавлено публичное свойство SelectedRadiocomponentTypeName

Commit:
feature(project MVVM, SearchRadiocomponentViewModel): привязано публичное свойство SelectedRadiocomponentTypeName к выбранному значению комбобокса

Commit:
feature(project MVVM, SearchRadiocomponentViewModel): добавлен параметр selectedObjects в конструктор

- добавлено приватное поле _selectedObjects, которому в конструкторе присваивается значение selectedObjects

Commit:
feature(project MVVM, SearchRadiocomponentViewModel): добавлен метод SelectFilteredRadiocomponents для обновления выделенных радиокомпонентов по результатам поиска

Commit:
refactor(project MVVM, SearchRadiocomponentViewModel): удалено поле _indexedRadiocomponents

- теперь в классе создано поле _radiocomponents, которому присваивается radiocomponents в конструкторе
- проиндексированные радиокомпоненты создаются непосредственно в методе SelectFilteredRadiocomponents

Commit:
feature(project MVVM, SearchRadiocomponentViewModel): добавлена команда SearchCommand

Commit:
feature(project MVVM, SearchRadiocomponentWindow.xaml): команда SearchCommand привязана к кнопке Найти

Commit:
feature(project MVVM, MainViewModel.OpenSearchWindowCommand): в конструктор searchRadiocomponentViewModel передается SelectedObjects

Commit:
fix(project MVVM, SearchRadiocomponentViewModel.GetFilteredByValueIndexedRadiocomponentsIndices): теперь при отключенном фильтре возвращается пустая коллекция

Commit:
feature(project MVVM, SearchRadiocomponentWindow.xaml): настроены параметры привязок чекбоксов

Commit:
fix(project MVVM, SearchRadiocomponentViewModel.SelectFilteredRadiocomponents): исправлена фильтрация радиокомпонентов по значению

- из метода GetFilteredByValueIndexedRadiocomponentsIndices удален параметр isComparatorTurnedOn
- проверка на то, включен ли фильтр и соответствующие действия теперь производятсяв в методе SelectFilteredRadiocomponents

Commit:
refactor(project MVVM, SearchRadiocomponentViewModel): переменная allTypesText вынесена как константа _allTypesText на уровень класса

Commit:
feature(project MVVM, AddRadiocomponentView.xaml): AddRadiocomponentWindow переименовано в ThisWindow

Commit:
refactor(project MVVM, AddRadiocomponentWindow): переименовано из AddRadiocomponentView

Commit:
fix(project MVVM, CustomDataGrid): в конструкторе добавлен костыль, снабженный комментарием, для корректного вытаскивания SelectedItems

- инструкция SelectedIndex = 0; принудительно выделяет первый элемент таблицы при первом добавлении новых элементов. Без этой строки привязанное к BindableSelectedItems свойство вьюмодели будет null, если пользователь еще ни разу не выделил ни одной строки, вместо пустой коллекции объектов

Commit:
refactor(project MVVM, SearchRadiocomponentViewModel.SelectFilteredRadiocomponents): удалена излишняя проверка перед пересечением коллекций

Commit:
feature(project MVVM): добавлена внешняя библиотека TemzzzWpfControlsSDK.dll

- создана папка Libs и в нее скопирован файл TemzzzWpfControlsSDK.dll
- зависимость обновлена зависимость TemzzzWpfControlsSDK в References

iKalentii commented on MVVM/CustomDataGrid.cs:
```diff
@@ -0,0 +1,85 @@
+    /// <see cref="DataGrid"/>, но имеющий возможность привязки к выделенным
+    /// элементам, чего нет в обычном <see cref="DataGrid"/>.
+    /// </summary>
+    public class CustomDataGrid : DataGrid
```
Вынести в библиотеку своих компонентов

iKalentii commented on MVVM/IActionViewModel.cs:
```diff
@@ -0,0 +1,38 @@
+    /// <summary>
+    /// Интерфейс модели представления действия с опциями.
+    /// </summary>
+    interface IActionViewModel : INotifyPropertyChanged
```
Тут корректнее было бы ActionWindow

iKalentii commented on MVVM/IndexToRadiocomponentTypeConverter.cs:
```diff
@@ -0,0 +1,67 @@
+    /// </summary>
+    public class IndexToRadiocomponentTypeConverter : IValueConverter
+    {
+        /// <inheritdoc cref="GetRadiocomponentTypeByIndex"/>
```
Не понял отсылку к GetRadiocomponentTypeByIndex

iKalentii commented on MVVM/RadiocomponentTypesToTypeAsStringToQuantityUnitAsStringMapConverter.cs:
```diff
@@ -0,0 +1,78 @@
+    /// единицы измерения.
+    /// </summary>
+    public class
+        RadiocomponentTypesToTypeAsStringToQuantityUnitAsStringMapConverter
```
Название О_О

iKalentii commented on MVVM/WindowCommands.cs:
```diff
@@ -0,0 +1,13 @@
+
+namespace MVVM
+{
+    public static class WindowCommands
```
Нигде не используется

TemZZZ replied:
Resolved.
Используется в XAML кнопками Отмена/Закрыть.

iKalentii commented on Model/IPrintableRadiocomponent.cs:
```diff
@@ -0,0 +1,30 @@
+    /// <summary>
+    /// Интерфейс удобочитаемого представления радиокомпонента.
+    /// </summary>
+    public interface IPrintableRadiocomponent
```
Как по мне, это ViewModel к радиокомпоненту, а не PrintableRadiocomponent

iKalentii commented on Testing/Model.UnitTests/RadiocomponentServiceTests.cs:
```diff
@@ -0,0 +1,164 @@
+using System.Collections.Generic;
+using NUnit.Framework;
+
+namespace Model.UnitTests
```
Не всё оттестировано в модели

![image](https://user-images.githubusercontent.com/902806/93340905-a908f080-f857-11ea-82bf-b89885e1293f.png)

Commit:
feature(project MVVM, CustomDataGrid): добавлена возможность задавать имена столбцов таблицы через атрибут  DescriptionAttribute свойств объектов

style+fix(project MVVM, CustomDataGrid): исправлены XML комментарии, удалены неиспользуемые директивы using

Commit:
feature(project Model, RadiocomponentToPrintableRadiocomponentAdapter): добавлены атрибуты Description свойствам

Commit:
refactor(project MVVM): переименован интерфейс IActionViewModel и реализующие его классы

- сам интерфейс IActionViewModel  переименован в IActionWindowViewModel
- ActionViewModelBase переименован в ActionWindowViewModelBase
- SaveToFileViewModel переименован в SaveToFileWindowViewModel
- LoadFromFileViewModel переименован в LoadFromFileWindowViewModel

Commit:
style(project MVVM, BoolToVisibilityConverter): добавлены XML комментарии

Commit:
style(project MVVM, IndexToRadiocomponentTypeConverter): исправлены XML комментарии, добавлен регион для статических методов

Commit:
style(project MVVM, RadiocomponentTypesToTypeAsStringToQuantityUnitAsStringMapConverter): исправлены XML комментарии, добавлен регион для статических методов

Commit:
refactor(project Model): интерфейс IPrintableRadiocomponent переименован в IRadiocomponentViewModel

- также переименован класс RadiocomponentToPrintableRadiocomponentAdapter в RadiocomponentToRadiocomponentViewModelAdapter

Commit:
feature(project Model, IRadiocomponentViewModel): добавлено свойство Radiocomponent

Commit:
refactor(project Model, RadiocomponentToRadiocomponentViewModelAdapter): метод GetRadiocomponent преобразован в свойство Radiocomponent

Commit:
feature(project MVVM): добавлен и реализован класс коллекции BindableCollection<T>

- можно свойству класса T назначить атрибут [Browsable(false)], и это свойство не будет отображаться в DataGrid

Commit:
refactor(project MVVM, MainViewModel.Radiocomponents): теперь это автосвойство инициализируется как пустая коллекция типа BindableCollection

Commit:
feature(project Model, RadiocomponentToRadiocomponentViewModelAdapter.Radiocomponent): добавлен атрибут [Browsable(false)]

Commit:
refactor(project Model, RadiocomponentToRadiocomponentViewModelAdapter): свойство Radiocomponent переделано в автосвойство

Commit:
feature(project MVVM, MainViewModel.cs, MainViewModel.xaml): реализован дизэйбл текстового поля вывода импеданса и выбор типа радиокомпонента, если выделен не один радиокомпонент

- в класс MainViewModel добавлено свойство IsSingleRadiocomponentSelected
- свойство IsSingleRadiocomponentSelected привязано к полю вывода импеданса и радиокнопкам изменения радиокомпонента

Commit:
refactor(project MVVM, MainWindow.xaml): столбцы таблицы теперь формируются вручную в XAML файле

Commit:
feature(project MVVM, MainWindow.xaml): добавлено правило валидации значения физической величины радиокомпонента в таблице

Commit:
fix(project MVVM, MainWindow.xaml): установлен режим привязки OneWay для столбцов таблицы Type, Quantity, Unit, чтобы не возникало ошибки при попытке изменить их пользователем

Commit:
refactor(poject MVVM, IDialogService): изменены сигнатуры методов OpenFileDialog и SaveFileDialog

- теперь методы принимают параметры string defaultExtension, string filter

Commit:
feature(project MVVM, DefaultDialogService): в методах OpenFileDialog и SaveFileDialog происходит назначение стандартного расширения файлов DefaultExt и фильтра Filter

Commit:
feature(project MVVM, SaveToFileWindowViewModel): диалоговому окну сохранения файла в команде ActionCommand передаются _defaultExtension и _filter

- сознады константы _filter и _defaultExtension

Commit:
feature(project MVVM, LoadFromFileWindowViewModel): диалоговому окну загрузки из файла в команде ActionCommand передаются _defaultExtension и _filter

- сознады константы _filter и _defaultExtension

Commit:
refactor(project MVVM, RadiocomponentTypesToStringsDictionaryConverter): переименован из RadiocomponentTypesToTypeAsStringToQuantityUnitAsStringMapConverter

Commit:
refactor(poject Model -> MVVM): перенесены IRadiocomponentViewModel и RadiocomponentToRadiocomponentViewModelAdapter из проекта Model в проект MVVM

Commit:
style(project View, SortableBindingList): немного изменен код, чтобы избавиться от директивы System.Windows.Forms

Commit:
refactor(project MVVM, MainWindow.xaml): компонент CustomDataGrid теперь достается из библиотеки TemzzzWpfControlsSDK.dll

Commit:
reafactor(project MVVM, CustomDataGrid): перенесен в библиотеку TemzzzWpfControlsSDL.dll

Commit:
feature(project MVVM, Libs/TemzzzWpfControlsSDK.dll): обновлена библиотека

- теперь она содержит CustomDataGrid

Commit:
feature(project Model.UnitTests, RadiocomponentServiceTests): добавлены и реализованы тестовые данные для методов ToString класса RadiocomponentService, а также тестовые методы

Commit:
feature(solution): создан проект RandomizersLib

- добавлен интерфейс IRandomizer

Commit:
feature(project RandomizerLib): добавлен и реализован класс стандартного рандомизатора DefaultRandomizer

Commit:
fix(project RandomizersLib, IRandomizer): добавлен модификатор доступа public

Commit:
feature(project RandomizersLib): добавлен и реализован класс GlobalRandomizer, выполняющий роль контекста окружения для классов рандомизаторов

Commit:
feature(project Model): добавлена ссылка на проект RandomizersLib

Commit:
refactor(project Model, RadiocomponentFactory.CreateRandomRadiocomponent): теперь рандомизатор назначается как GlobalRandomizer.Instance

Commit:
feature(project Model.UnitTests): добавлена ссылка на проект RandomizersLib

Commit:
feature(project Model.UnitTests): добавлен и реализован класс FakeRandomizer

Commit:
feature(project Model.UnitTests, RadiocomponentFactoryTests): добавлен конструктор, в котором GlobalRandomizer.Instance назначается экземпляр FakeRandomizer для нужд тестирования

Commit:
fix(project RandomizersLib, GlobalRandomizer): сеттер свойства Instance сделан публичным, чтобы иметь к нему доступ в проекте с юнит-тестами

Commit:
style(project Model.UnitTests, RadiocomponentFactoryTests): код разделен на регионы

Commit:
test(project Model.UnitTest, RadiocomponentFactoryTests): добавлены тестовые данные и тесты для метода CreateRandomRadiocomponent

Commit:
feature(Libs\TemzzzWpfControlsSDK.dll): пересобрана библиотека

Commit:
refactor(project MVVM): файлы вьюмоделей перенесены в папку VMs

Commit:
refactor(project MVVM): все представления (XAML файлы, кроме App.xaml) перенесены в папку Views

Commit:
refactor(project MVVM): все классы конвертеров перенесены в папку Converters

Commit:
refactor(project MVVM, BrowsableObservableCollection): переименован из BindableCollection

Commit:
refactor(project Model): классы Resistor, Inductor, Capacitor перенесены из пространства имен Model.PassiveComponents в Model

- пространства имен Model.PassiveComponents больше не существует

Commit:
fix(project Model, RadiocomponentBase.cs): удалена несуществующая директива using Model.PassiveComponents;

Commit:
refactor(solution): проект RandomizersLib и пространство имен RandomizersLib переименованы в Randomizers

Commit:
refactor(project MVVM, GlobalRandomizer): класс сделан статическим

Commit:
refactor(project MVVM, RandomizersAmbientContext): переименован из GlobalRandomizer

Commit:
refactor(project MVVM, folder VMs): все включения ViewModel  в именах файлов, классов, переменных заменены на VM

Commit:
fix(project MVVM, PresentationRootRegistry.CheckViewModelTypeIsNotInterface): исправлен XML комментарий

Commit:
refactor(project MVVM, App.App): mainViewModel переименована в mainVM

Commit:
refactor(project MVVM, MainVM): включения ViewModel в именах переменных в командах заменены на VM

Commit:
refactor(project MVVM, ViewRootRegistry): переименован из PresentationRootRegistry

Commit:
refactor(project MVVM, RadiocomponentVM): переименован из RadiocomponentToRadiocomponentVMAdapter

- класс RadiocomponentVM больше не реализует интерфейс IRadiocomponentVM

Commit:
feature(project MVVM): удален интерфейс IRadiocomponentVM за ненадобностью

Commit:
refactor(project MVVM, MainVM.Radiocomponents): изменен тип коллекции с BrowsableObservableCollection на ObservableCollection

Commit:
feature(project MVVM, MainVM): частоте назначено значение по умолчанию _frequencyAsString = "0";

Commit:
refactor(project View --> project Model): перенесены некоторые файлы из проекта View в проект Model

- RadiocomponentsLoadOption.cs
- RadiocomponentsSaveOption.cs
- XmlReaderWriter.cs
- ExceptionHandler.cs

Commit:
feature(project MVVM, MainWindow.xaml): реализовано сокрытие наименования физической величины и единицы измерения в области "Изменить радиокомпонент"

- реализован триггер, реагирующий на свойство IsEnabled элемента RadioButtonSelector, и меняющий его свойство IsReadOnlyTextBlockVisible на false

Commit:
refactor(project MVVM, IDialogService): переименован из IDIalogService

Commit:
refactor(project MVVM): файлы NotNegativeDoubleValidationRule.cs и ValidationRuleService.cs перенесены в папку ValidationRules

Commit:
refactor(project Model, Resistor.cs, Inductor.cs, Capacitor.cs): убраны константы zeroImaginaryPart и zeroRealPart

Commit:
feature(project Model, RadiocomponentBase): удален конструктор без параметров

Commit:
feature(project Model, RadiocomponentBase): удален конструктор без параметров

Commit:
refactor(project Model, Resistor, Inductor, Capacitor): конструктор без параметров вызывает конструктор базового класса с передачей в него значения по умолчанию

Commit:
style(project MVVM, MainWindow.xaml): добавлены имена свойств Path в биндингах, исправлено выравнивание кода

Commit:
feature(project MVVM, NotNegativeDoubleValidationRule.Validate): переведено на русский сообщение об ошибке ввода

Commit:
feature(project MVVM, MainWindow.xaml): добавлено всплывающее сообщение о некорректном вводе чилсового значения в ячейку

Commit:
fake-refactor(solution): все файлы, которые на GitHub'е имели в имени ...Control... с большой буквы, переименованы в ...control... с маленькой буквы и с суффиксом _

Commit:
fake-refactor(solution): удален RadioComponentBase.cs (Control c большой буквы)

Commit:
fake-refactor(solution): RadioComponentsCollection.sln и RadioComponentsCollection.sln.DotSettings переименованы в RadiocomponentsCollection_.sln и RadiocomponentsCollection_.sln.DotSettings соответственно

Commit:
fake-refactor(solution): удален файл RadioComponentsCollection.sln.DotSettings

Commit:
fake-rafactor(solution): в именах всех файлов с суффиксами _ удалены суффиксы _

Commit:
feature(project Model): добавлена NuGet библиотека Newtonsoft Json

- добавлен и реализован класс ChildrenTypesSerializationBinder, который будет использоваться для сериализации

Commit:
feature(project Model, XmlReaderWriter): добавлен и реализован метод GetStreamWriter для создания потока записи в текстовый файл

Commit:
style+refactor(project Model, XmlReaderWriter): StreamExceptionTypeToMessageMap переименован в StreamExceptionTypeToMessageDictionary

- также немного изменено форматирование словаря StreamExceptionTypeToMessageDictionary
- в XML комментариях класса в конце предложений поставлены точки

Commit:
refactor(project Model, XmlReaderWriter): свойство StreamExceptionTypeToMessageDictionary переделано в приватное поле только для чтения _streamExceptionTypeToMessageDictionary

Commit:
refactor(project Model): переименованы методы

- SerializeAndWriteXml в SerializeAndWriteToFile
- ReadXmlAndDeserialize в ReadFileAndDeserialize

Commit:
refactor(project Model, Serializer): переименован из XmlReaderWriter

Commit:
style(project Model, Serializer): код разделен на регионы

Commit:
refactor(project Model, FilesReaderWriter): переименован из Serializer

- из XML комментариев удалены упоминания об XML

Commit:
feature(project Model): добавлен и описан интерфейс сериализатора ISerializer

Commit:
feature(project Model, ISerializer): изменены сигнатуры методов Serialize и Deserialize

Commit:
feature(project Model): добавлен и реализован класс CustomJsonSerializer

Commit:
feature(project Model, FilesReaderWriter): добавлен и реализован конструктор, принимающий сериализатор

- сериализатор хранится в приватном поле _serializer
- теперь в методах SerializeAndWriteToFile и ReadFileAndDeserialize вместо serializer и deserializer используется _serializer
- в методе SerializeAndWriteToFile в переменную file значение возвращается методом GetStreamWriter

Commit:
refactor(project View.MainForm, project MVVM.LoadFromFileWindowVM, project MVVM.SaveToFileWindowVM): объектам FilesReaderWriter передаются сериалиаторы в качестве параметра

Commit:
feature(project Model, CustomJsonSerializer): добавлен и реализован конструктор, реализовано сохранение имен типов сериализуемых объектов

- инициализация _serializer теперь происходит в конструкторе
- в конструкторе свойству TypeNameHandling объекта-сериализатора присваивается значение TypeNameHandling.All

Commit:
style(project View): классы, лежащие в папке EventArgsClasses, теперь принадлежат пространству имен View.EventArgsClasses, а лежащие в папке Forms - View.Forms

Commit:
fix(project Model, CustomJsonSerializer): изменен тип TypeNameHandling.All на TypeNameHandling.Objects в конструкторе

Commit:
refactor(project Model, CustomJsonSerializer, ChildrenTypesSerializationBinder): using'и, содержащие Newtonsoft.Json, вынесены из namespace Model

Commit:
refactor(project Model): файлы ISerializer.cs, CustomJsonSerializer.cs, ChildrenTypesSerializationBinder.cs перенесены в папку Serializers

Commit:
refactor(project Model, FilesReaderWriter.SerializeAndWriteToFile): serializableObject переименован в serializingObject

Commit:
refactor(FilesReaderWriter): serializationErrorText и deserializationErrorText переименованы в serializationErrorMessage и deserializationErrorMessage соответственно

Commit:
feature(project Model, FilesReaderWriter): реализован повторный выброс исключения в методах SerializeAndWriteToFile и ReadFileAndDeserialize, если errorMessager не назначен

Commit:
feature(project Model): добавлен и реализован класс RadiocomponentsLoaderSaver с логикой загрузки/сохранения радиокомпонентов из файла/в файл

Commit:
refactor(project Model, TextFilesReaderWriter): переименован из FilesReaderWriter

Commit:
feature(project Model, TextFilesReaderWriter): удален приватный метод GetFileStream не относящийся к задачам класса

Commit:
feature(project Model, RadiocomponentsLoaderSaver): коллекции теперь передаются не в конструктор, а непосредственно в методы

Commit:
refactor(project MVVM, LoadFromFileWindowVM): команда загрузки радиокомпонентов переписана через использование RadiocomponentsLoaderSaver

Commit:
refactor(project MVVM, LoadFromFileWindowVM.ToRadiocomponentVMs): переименован из ToRadiocomponentVMs

Commit:
refactor(project MVVM, LoadFromFileWindowVM.ActionCommand): цикл в команде переписан через вызовы методов AddItems и ToRadiocomponentVMs

Commit:
feature+refactor(project MVVM, LoadFromFileWindowVM): переименовано поле _radiocomponents в _radiocomponentVMs

- LINQ выражение в команде ActionCommand для получения радиокомпонентов из вьюмоделей заменено методом GetRadiocomponents

Commit:
refactor(project Model, RadiocomponentsReaderWriter): переименован из RadiocomponentsLoaderSaver

Commit:
feature(project MVVM): добавлен и реализован класс ObservableViewModelCollection, представляющий синхронизированную с коллекцией моделей коллекцию вьюмоделей

Commit:
refactor(project MVVM, SyncObservableViewModelCollection): переименован из ObservableViewModelCollection

Commit:
feature(project MVVM, MainVM): добавлена приватная коллекция _radiocomponents, которая будет содержать модели радиокомпонентов

Commit:
refactor(project MVVM, MainVM.RadiocomponentVMs): переименовано из Radiocomponents

Commit:
feature(): проинициализированы коллекция _radiocomponents на уровне класса и RadiocomponentVMs в конструкторе

Commit:
feature(project MVVM, MainVM): вместо коллекции RadiocomponentVMs теперь используется _radiocomponents

Commit:
refactor(project MVVM, MainVM): изменения в коде класса

- изменено имя поля _singleSelectedRadiocomponentVM на _singleSelectedRadiocomponent
- изменен тип поля _singleSelectedRadiocomponent с RadiocomponentVM на RadiocomponentBase
- в методе UpdateSelectedRadiocomponentImpedanceAsString метод GetImpedance теперь напрямую вызывается у _singleSelectedRadiocomponent
- в методе UpdateSelectedRadiocomponentTypeIndex свойство Type теперь напрямую вызывается у _singleSelectedRadiocomponent
- в свойстве SelectedObjects полю _singleSelectedRadiocomponent присваивается значение свойства Radiocomponent
- в команде OpenAddRadiocomponentWindowCommand добавлено получение коллекции выделенных радиокомпонентов selectedRadiocomponents
- в команде ModifyRadiocomponentCommand по индексу идет обращение к коллекции RadiocomponentVMs
- в команде ModifyRadiocomponentCommand переменная selectedPrintableRadiocomponents переименована в selectedRadiocomponents, а ее тип теперь List<RadiocomponentBase>
- в команде ModifyRadiocomponentCommand изменено пополнение коллекции selectedRadiocomponents

Commit:
refactor(project MVVM): переименованы окна и вьюмодели

- AddRadiocomponentWindow в AddingRadiocomponentWindow
- SearchRadiocomponentWindow в SearchingRadiocomponentWindow
- AddRadiocomponentVM в AddingRadiocomponentVM
- LoadFromFileWindowVM в LoadingFromFileWindowVM
- SaveToFileWindowVM в SavingToFileWindowVM
- SearchRadiocomponentVM в SearchingRadiocomponentVM

Commit:
refactor(project MVVM, MainVM): удален метод ToPrintableRadiocomponents, так как он нигде не используется

Commit:
refactor(project MVVM, MainVM): небольшие переименования переменных в командах

- именам объектов-окон и объектов-вьюмоделей добавлены -ing окончания

Commit:
refactor(project MVVM, MainVM): переименованы команды

- именам команд, начинающихся с Open, добавлены -ing окончания

Commit:
refactor(project Model, RadiocomponentsReaderWriter): в методах SaveToFile и LoadFromFile изменены типы входных параметров

- в методе LoadFromFile метод AddRange заменен циклом

Commit:
refactor(project MVVM, AddingRadiocomponentVM): изменения в коде

- коллекция _radiocomponents теперь имеет тип ICollection<RadiocomponentBase>
- в методе AddRadiocomponent переменная printableRadiocomponent переименована в radiocomponent
- в конструкторе тип параметра radiocomponents теперь ICollection<RadiocomponentBase>
- _adddRadiocomponentCommand переименована в _addRadiocomponentCommand

Commit:
refactor(project MVVM, LoadingFromFileWindowVM): изменения в коде

- тип _radiocomponents теперь IList<RadiocomponentBase>
- команда _openLoadFromFileDialogCommand переименована в _openLoadingFromFileDialogCommand
- удалены методы ToRadiocomponentVMs, GetRadiocomponents, AddItems за ненадобностью
- параметр конструктора radiocomponentVMs переименован в radiocomponents, также теперь он имеет тип IList<RadiocomponentBase>
- в ActionCommand fileReader переименован в textFileReader, radiocomponentsLoader переименован в radiocomponentsReader
- в ActionCommand удалены лишние преобразования методами GetRadiocomponents, AddItems, ToRadiocomponentVMs

Commit:
refactor(project MVVM, SavingToFileWindowVM): изменения в коде

- поля _radiocomponents и _selectedRadiocomponents теперь имеют тип IList<RadiocomponentBase>
- команда _openLoadFromFileDialogCommand переименована в _openSavingToFileDialogCommand
- удален метод GetWritingRadiocomponents за ненадобностью
- изменены типы параметров конструктора на IList<RadiocomponentBase>
- вся ответственность за сохранение в файл в команде ActionCommand отдана объекту radiocomponentsWriter

Commit:
refactor(project MVVM, SearchingRadiocomponentVM): изменения в коде

- поле _radiocomponents теперь имеет тип IList<RadiocomponentBase>
- исправлены XML комментарии по всему классу
- метод GetIndexedRadiocomponents теперь возвращает List<(int, RadiocomponentBase)>
- метод GetFilteredByTypeIndexedRadiocomponents теперь принимает indexedRadiocomponents типа IEnumerable<(int, RadiocomponentBase)>
- метод GetFilteredByValueIndexedRadiocomponentsIndices  теперь принимает indexedRadiocomponents типа IEnumerable<(int, RadiocomponentBase)>
- тип коллекции filteredByTypeIndexedRadiocomponents в методе SelectFilteredRadiocomponents изменен на List<(int, RadiocomponentBase)>
- конструктор теперь принимает параметр radiocomponents типа IList<RadiocomponentBase>

Commit:
feature(project MVVM, SyncObservableViewModelCollection.OnModelsCollectionChanged): изменена логиа работы метода с учетом особенностей ObservableCollection

Commit:
style(project MVVM, MainVM.cs): удалена неиспользуемая директива using

Commit:
fix(project MVVM, MainVM): исправлена ошибка: импеданс не считался до изменения пользователем значения частоты

- свойство FrequencyAsString инициализируется значением по умолчанию в конструкторе

Commit:
feature(project MVVM, MainWindow.xaml): добавлен регулирующий ширину колонок разделитель между таблицей и областью с кнопками

Commit:
feature(project MVVM, MainVM): добавлен и реализован метод ComplexToString

- в методе UpdateSelectedRadiocomponentImpedanceAsString свойству SelectedRadiocomponentImpedanceAsString присваивается результат, возвращаемый методом ComplexToString

Commit:
fix(project MVVM, SavingToFileWindowVM.ActionCommand): исправлена ошибка: если не было выделено ни одного радиокомпонента в таблице, то нельзя было сохранить даже с опцией "Все радиокомпоненты"

- добавлена проверка выделенной опции перед вызовом диалога сохранения файла

Commit:
feature(project Model, RadiocomponentFactory.CreateRandomRadiocomponent): словарь typeToDivisorMap переделан в typeToMultiplierDictionary

Commit:
refactor(project Model, RadiocomponentFactory.CreateRandomRadiocomponent): typeToMultiplierDictionary переименован в typeToMultiplierTuples

Commit:
refactor(project Model, RadiocomponentFactory.CreateRandomRadiocomponent): typeToMultiplierTuples переименован в _typeToMultiplierTuples и вынесен из метода на уровень класса

Commit:
refactor(project Model, RadiocomponentFactory.GetRandomRadiocomponent): переименован из CreateRandomRadiocomponent

Commit:
refactor(project Model, RadiocomponentFactory.GetRadiocomponent): переименован из CreateRadiocomponent

Commit:
refactor(project Model.UnitTests, RadiocomponentFactoryTests): небольшие переименования

- _radiocomponentInfoDictionary переименован в _radiocomponentInfoTuples
- CreateRadiocomponentTestCases переименован в GetRadiocomponent_TestCasesData
- GetRadiocomponentTypeTestCases переименован в GetRadiocomponentType_TestCasesData
- CreateRandomRadiocomponent_NoParameters_TestCasesData переименован в GetRandomRadiocomponent_NoParameters_TestCasesData
- CreateRadiocomponent_ReceivedGoodValues_ReturnsObjectOfExpectedType переименован в GetRadiocomponent_ReceivedGoodValues_ReturnsObjectOfExpectedType
- CreateRandomRadiocomponent_NoParameter_ReturnsValue переименован в GetRandomRadiocomponent_NoParameter_ReturnsValue
- имена методов в именах тестовых случаев записаны через nameof

Commit:
refactor+style(project Model, RadiocomponentFactory.GetTypeToRadiocomponentDictionary): переименован из GetTypeToRadiocomponentMap

- добавлены точки в конце предложений в XML комментариях

Commit:
feature+style(project Model, IRadiocomponent): добавлено свойство Type типа RadiocomponentType

- изменен порядок следования свойств
- добавлены точки в конце XML комментариев

Commit:
style(project Model, RadiocomponentBase): оформлен код

- удалены атрибуты XmlInclude за ненадобностью
- добавлены точки в конце XML комментариев

Commit:
refactor(project Model, RadiocomponentBase.CalculateAndGetImpedance): переименован из CalculateImpedance

Commit:
style(project Model, RadiocomponentBase): код сгруппирован по регионам

Commit:
feature(project Model, RadiocomponentBase): свойства Quantity и Unit сделаны абстрактными

Commit:
feature(project Model): в классах Resistor, Inductor, Capacitor реализованы свойства Quantity и Unit

Commit:
refactor(project MVVM, RadiocomponentTypesToPropertiesTuplesConverter): переименован из RadiocomponentTypesToStringsDictionaryConverter

- GetRadiocomponentTypeAsStringToQuantityUnitAsStringMap переименован в GetRadiocomponentPropertiesTuples

style(solution): в XML комментариях добавлены точки в конце предложений

refactor(solution): методы, переменные, поля, содержащие в имени Map, переименованы в более соответствующие им сущности Dictionary или Tuples

Commit:
refactor(project MVVM, MainWindow.xaml): свойство RadiocomponentTypeAsStringToQuantityUnitAsStringMap переименовано в RadiocomponentPropertiesTuples

Commit:
refactor(project MVVM, AddingRadiocomponentWindow.xaml): свойство RadiocomponentTypeAsStringToQuantityUnitAsStringMap переименовано в RadiocomponentPropertiesTuples

Commit:
feature(project MVVM, Libs): обновлен TemzzzWpfControlSDK.dll

Commit:
feature(project MVVM, Libs): обновлен файл TemzzzWpfControlsSDK.dll

Commit:
feature(project MVVM, Views): переименовано свойство

- в файлах ActionWindow.xaml, AddingRadiocomponentWindow.xaml, MainWindow.xaml переименовано свойство RadioButtonTextToReadOnlyTextMap в RadioButtonTextToReadOnlyTextTuples

Commit:
fix(project MVVM, LoadingFromFileWindowVM): исправлена ошибка: при попытке загрузить файл неверного формата выбрасывалось необработанное исключение

- в ActionCommand добавлен отлов исключения JsonReaderException

Commit:
style(project MVVM, ActionWindowVMBase): вместо inheritdoc-комментариев написаны развернутые XML комментарии

Commit:
feature(project MVVM, ActionWindowVMBase): удален интерфейс IActionWindowVM из перечня реализуемых классом

Commit:
feature(project MVVM): удален интерфейс IActionWindowVM за ненадобностью

Commit:
refactor(project Model, LoadOption): переименован из RadiocomponentsLoadOption

Commit:
refactor(project Model, SaveOption): переименован из RadiocomponentsSaveOption

Commit:
refactor(project MVVM, SavingRadiocomponentsWindowVM): переименован из SavingToFileWindowVM

Commit:
refactor(project MVVM, LoadingRadiocomponentsWindowVM): переименован из LoadingFromFileWindowVM

Commit:
feature(project MVVM, RadiocomponentsReaderWriter): добавлены константы и свойства, связанные с загрузкой из файла и сохранением в файл

- добавлена константа DefaultExtension, хранящая стандартное расширение файлов радиокомпонентов
- добавлена константа DefaultFilesFilter, хранящая стандартный фильтр файлов
- добавлены статические свойства SaveOptionToDescriptionDictionary и LoadOptionToDescriptionDictionary

Commit:
refactor(project MVVM, LoadingRadiocomponentsWindowVM): удалены поля _defaultExtension, _filter, _loadingOptionToDescriptionDictionary

- вместо них в классе используются RadiocomponentsReaderWriter.DefaultExtension, RadiocomponentsReaderWriter.DefaultFilesFilter, RadiocomponentsReaderWriter.LoadOptionToDescriptionDictionary

Commit:
refactor(project MVVM, SavingRadiocomponentsWindowVM): удалены поля _defaultExtension, _filter, _loadingOptionToDescriptionDictionary

- вместо них в классе используются RadiocomponentsReaderWriter.DefaultExtension, RadiocomponentsReaderWriter.DefaultFilesFilter, RadiocomponentsReaderWriter.LoadOptionToDescriptionDictionary

Commit:
feature(project Model, RadiocomponentService): изменен и переименован словарь _radiocomponentTypeToPropertiesDictionary

- словарь переименован в _radiocomponentTypesInfoDictionary
- теперь в словаре дополнительно содержится информация о сокрываемом типе радиокомпонента (Resistor, Capacitor, Inductor)
- реализован метод RadiocomponentType GetRadiocomponentType(IRadiocomponent radiocomponent)
- реализован метод List<RadiocomponentType> GetAvailableRadiocomponentTypes()
- в методе ToString(RadiocomponentType radiocomponentType, double radiocomponentValue) значения quantity и unit достаются из словаря _radiocomponentTypesInfoDictionary

Commit:
refactor(project MVVM, MainVM): список _availableRadiocomponentTypes теперь берется из RadiocomponentService.GetAvailableRadiocomponentTypes()

Commit:
refactor(project View, класс RadiocomponentControl, сеттер свойства Radiocomponent): radiocomponentType теперь берется из RadiocomponentService.GetRadiocomponentType

Commit:
refactor(project Model, RadiocomponentFactory): удален словарь _typeToRadiocomponentTypeDictionary и метод GetRadiocomponentType, т. к. они перенесены в класс RadiocomponentService

Commit:
feature(project Model.UnitTests, RadiocomponentFactoryTests): закомментированы тестовые случаи GetRadiocomponentType_TestCasesData и тестовый метод GetRadiocomponentType_ReceivedGoodValue_ReturnsValue

Commit:
feature(project RadiocomponentService): добавлен и реализован метод GetEncapsulatedType

refactor(project RadiocomponentService): параметры методов type переименованы в radiocomponentType

Commit:
refactor(project Model, RadiocomponentsFactory.GetRadiocomponent): метод переписан через RadiocomponentService.GetEncapsulatedType и Activator.CreateInstance

Commit:
feature(project Model, RadiocomponentFactory): удален метод GetTypeToRadiocomponentDictionary за ненадобностью

Commit:
feature(project Model, RadiocomponentFactory.GetRandomRadiocomponent): вместо _typeToMultiplierTuples используется RadiocomponentService.RadiocomponentTypeToMultiplierTuplesForRandom

Commit:
feature(project Model, RadiocomponentService): добавлено свойство RadiocomponentTypeToMultiplierTuplesForRandom

Commit:
style(project Model, RadiocomponentService): добавлены XML коментарии к методам и полям _radiocomponentTypesInfoDictionary, GetAvailableRadiocomponentTypes, GetEncapsulatedType

Commit:
feature(project Model, RadiocomponentFactory): удален список _typeToMultiplierTuples за ненадобностью

Commit:
fix(project MVVM, SearchingRadiocomponentVM.GetFilteredByValueIndexedRadiocomponentsIndices): удален лишний параметр из XML комментария

Commit:
refactor(project MVVM, SearchingRadiocomponentsVM): переименован из SearchingRadiocomponentVM

Commit:
refactor(project MVVM, MainVM.OpenSearchingRadiocomponentsWindowCommand): переименована из OpenSearchingRadiocomponentWindowCommand

Commit:
refactor(project MVVM, SearchingRadiocomponentsWindow): переименован из SearchingRadiocomponentWindow

Commit:
fix(project MVVM, MainWindow.xaml): свойство OpenSearchingRadiocomponentWindowCommand переименовано в OpenSearchingRadiocomponentsWindowCommand

Commit:
fix(project MVVM): исправлена ошибка: не выделялись найденные радиокомпоненты

- в классе SearchingRadiocomponentsVM создано поле _radiocomponentVMs, значение которому передается в конструкторе класса
- в классе MainVM в команде OpenSearchingRadiocomponentsWindowCommand при создании экземпляра searchingRadiocomponentsVM в конструктор дополнительно передается коллекция RadiocomponentVMs
- в классе SearchingRadiocomponentsVM в методе SelectFilteredRadiocomponents в коллекцию _selectedObjects добавляются вьюмодели радиокомпонентов

Commit:
feature(project Model, ExceptionHandler): класс сделан статическим

Commit:
refactor(project Model, TextFilesReaderWriter): переименовано поле _streamExceptionTypeToMessageDictionary в _fileIoExceptionTypeToErrorMessageDictionary

Commit:
feature(project Model): добавлен и реализован публичный статический класс TextFileIOService

- в класс скопирован словарь _fileIoExceptionTypeToErrorMessageDictionary, метод GetStreamWriter, метод GetStreamReader из класса TextFilesReaderWriter

Commit:
refactor(proejct Model, TextFilesReaderWriter): в методах ReadFileAndDeserialize и SerializeAndWriteToFile используются TextFileIOService.GetStreamReader и TextFileIOService.GetStreamWriter вместо местных приватных одноименных методов

- удалены приватные методы GetStreamWriter, GetStreamReader за ненадобностью
- удалено приватное поле _fileIoExceptionTypeToErrorMessageDictionary за ненадобностью

Commit:
feature(project Model): класс TextFilesReaderWriter разделен на два класса

- добавлен и реализован класс TextFileReader, в который скопирован метод ReadFileAndDeserialize из класса TextFilesReaderWriter
- добавлен и реализован класс TextFileWriter, в который скопирован метод SerializeAndWriteToFile из класса TextFilesReaderWriter

Commit:
feature(project MVVM, LoadingRadiocomponentsWindowVM.ActionCommand): переменная textFileReader теперь является объектом класса TextFileReader

Commit:
feature(project MVVM, SavingRadiocomponentsWindowVM.ActionCommand): переменная textFileWriter теперь является объектом класса TextFileWriter

Commit:
refactor(project View, MainForm): изменения в методах LoadFromFile и SaveToFile

- в методе SaveToFile переменная fileWriter теперь является объектом класса TextFileWriter
- в методе LoadFromFile переменная fileReader теперь является объектом класса TextFileReader

Commit:
style(project Model, TextFileIOService): изменен XML комментарий

Commit:
feature(project Model): добавлен и реализован класс RadiocomponentsIOService

- в класс скопированы DefaultExtension, DefaultFilesFilter, SaveOptionToDescriptionDictionary, LoadOptionToDescriptionDictionary из класса RadiocomponentsReaderWriter

Commit:
refactor(project MVVM, LoadingRadiocomponentsWindowVM.GetOptionToDescriptionDictionary): теперь метод возвращает RadiocomponentsIOService.LoadOptionToDescriptionDictionary

Commit:
refactor(project MVVM, SavingRadiocomponentsWindowVM.GetOptionToDescriptionDictionary): теперь метод возвращает RadiocomponentsIOService.LoadOptionToDescriptionDictionary

Commit:
feature(project MVVM, RadiocomponentsReaderWriter): удалены DefaultExtension, DefaultFilesFilter, SaveOptionToDescriptionDictionary, LoadOptionToDescriptionDictionary за ненадобностью

Commit:
feature(project MVVM): класс RadiocomponentsReaderWriter разделен на два класса

- добавлен и реализован класс RadiocomponentsReader, в который скопирован метод LoadFromFile из класса RadiocomponentsReaderWriter
- добавлен и реализован класс RadiocomponentsWriter, в который скопирован метод SaveToFile из класса RadiocomponentsReaderWriter

Commit:
refactor(project MVVM, LoadingRadiocomponentsWindowVM.ActionCommand): переменная radiocomponentsReader теперь является объектом класса RadiocomponentsReader

Commit:
refactor(project MVVM, SavingRadiocomponentsWindowVM.ActionCommand): переменная radiocomponentsWriter теперь является объектом класса RadiocomponentsWriter

Commit:
feature(project Model): удалены классы RadiocomponentsReaderWriter и TextFilesReaderWriter за ненадобностью

Commit:
refactor(project Model): создана папка Services, в которую перенесены файлы

- ExceptionHandler.cs
- RadiocomponentService.cs
- RadiocomponentsIOService.cs
- TextFileIOService.cs

Commit:
refactor(project Model): создана папка IO, в которую перенесены файлы

- LoadOption.cs
- RadiocomponentsReader.cs
- RadiocomponentsWriter.cs
- SaveOption.cs
- TextFileReader.cs
- TextFileWriter.cs

Commit:
style(project Model, RadiocomponentsWriter): добавлен  XML комментарий

Commit:
refactor(project MVVM): файлы распределены по папкам

- в папку Views перенесен файл ViewRootRegistry.cs
- в папку VMs перенесены файлы BrowsableObservableCollection.cs, RelayCommand.cs, SyncObservableViewModelCollection.cs, WindowCommands.cs
- в созданную папку Services перенесены IDialogService.cs и DefaultDialogService.cs

Commit:
refactor(project MVVM): файл ValidationRuleService.cs перенесен из папки ValidationRules в папку Services

Commit:
refactor(project Model.UnitTests): создана папка Services и в нее перенесен файл RadiocomponentServiceTests.cs
